src/scene/scene_lua.inl

```diff
@@ -140,63 +141,64 @@ std::vector<openspace::properties::Property*> findMatchesInAllProperties(
         }
     }
 
-    // Stores whether we found at least one matching property. If this is false at the end
-    // of the loop, the property name regex was probably misspelled.
-    for (properties::Property* prop : properties) {
-        // Check the regular expression for all properties
-        const std::string id = prop->uri();
-
-        if (isLiteral && id != propertyName) {
-            continue;
-        }
-        else if (!propertyName.empty()) {
-            size_t propertyPos = id.find(propertyName);
-            if (propertyPos != std::string::npos) {
-                // Check that the propertyName fully matches the property in id
-                if ((propertyPos + propertyName.length() + 1) < id.length()) {
-                    continue;
-                }
-
-                // Match node name
-                if (!nodeName.empty() && id.find(nodeName) == std::string::npos) {
-                    continue;
+    std::mutex mutex;
+    std::for_each(
+        std::execution::par_unseq,
+        properties.cbegin(),
+        properties.cend(),
+        [&](properties::Property* prop) {
+            // Check the regular expression for all properties
+            const std::string_view id = prop->uri();
+
+            if (isLiteral && id != propertyName) {
+                return;
+            }
+            else if (!propertyName.empty()) {
```

Comment:
I can see that this hasn't changed from the previous implementation, but there is a lot of confusion in here to me.... Can we maybe rewrite it or rename some variables to make it clearer what is going on? E.g: 

1. `prop->uri()` is actually the full URI of the property, but here its' stored in a variable `id` which makes one think it's the identifier. How about renaming that variable to `uri`?
2. What is `propertyName`? It's not the name of the property, since it seems like here we compare it again `id`, which is really the URI.  Should `noedName` and `propertyName` above be renamed to something else, that more clearly describe what they are (e.g. name, identifier, or URI)?

Reply:
For 1:  Done
For 2:  I'd prefer to do that change in a second PR instead, including a bunch of new unit tests to validate that everything works.  There is a bunch of changes that doesn't make too much sense as is, but adding that in here would dilute this PR a lot

Reply:
https://github.com/OpenSpace/OpenSpace/issues/3588